### PR TITLE
python3Packages.tokenizers: 0.9.4 -> 0.10.0

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Changelog:
https://github.com/huggingface/tokenizers/releases/tag/python-v0.10.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
seems to fail for me locally:
```
https://github.com/NixOS/nixpkgs/pull/109532

4 packages failed to build:
python37Packages.tokenizers python37Packages.transformers python38Packages.tokenizers python38Packages.transformers
```

```
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/os.py:213: in makedirs
    makedirs(head, exist_ok=exist_ok)
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/os.py:223: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
```

File: pkgs/development/python-modules/tokenizers/default.nix [hunk 1]
```diff
@@ -105,6 +107,11 @@ in rustPlatform.buildRustPackage rec {
     pipInstallPhase
   '';
 
+  disabledTests = [
```
Comment (jonringer) [inline pkgs/development/python-modules/tokenizers/default.nix:110]:
```suggestion
  preCheck = ''
    HOME=$TMPDIR
  '';
  disabledTests = [
```

```
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/os.py:213: in makedirs
    makedirs(head, exist_ok=exist_ok)
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/os.py:223: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 109532` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python37Packages.tokenizers</li>
    <li>python37Packages.transformers</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 109532` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python37Packages.tokenizers</li>
    <li>python37Packages.transformers</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
  </ul>
</details>

```
rootdir: /private/tmp/nix-build-tokenizers-0.10.0.drv-0/source/bindings/python
collected 117 items / 1 error / 116 selected

==================================== ERRORS ====================================
__ ERROR collecting tests/documentation/test_tutorial_train_from_iterators.py __
tests/documentation/test_tutorial_train_from_iterators.py:4: in <module>
    import datasets
/nix/store/janmd6006jr9kmxipkc22g0g09b9j0n7-python3.8-datasets-1.1.2/lib/python3.8/site-packages/datasets/__init__.py:27: in <module>
    from .arrow_dataset import Dataset, concatenate_datasets
/nix/store/janmd6006jr9kmxipkc22g0g09b9j0n7-python3.8-datasets-1.1.2/lib/python3.8/site-packages/datasets/arrow_dataset.py:40: in <module>
    from .arrow_reader import ArrowReader
/nix/store/janmd6006jr9kmxipkc22g0g09b9j0n7-python3.8-datasets-1.1.2/lib/python3.8/site-packages/datasets/arrow_reader.py:31: in <module>
    from .utils import cached_path, logging
/nix/store/janmd6006jr9kmxipkc22g0g09b9j0n7-python3.8-datasets-1.1.2/lib/python3.8/site-packages/datasets/utils/__init__.py:20: in <module>
    from .download_manager import DownloadManager, GenerateMode
/nix/store/janmd6006jr9kmxipkc22g0g09b9j0n7-python3.8-datasets-1.1.2/lib/python3.8/site-packages/datasets/utils/download_manager.py:25: in <module>
    from .file_utils import HF_DATASETS_CACHE, cached_path, get_from_cache, hash_url_to_filename
/nix/store/janmd6006jr9kmxipkc22g0g09b9j0n7-python3.8-datasets-1.1.2/lib/python3.8/site-packages/datasets/utils/file_utils.py:118: in <module>
    os.makedirs(HF_MODULES_CACHE, exist_ok=True)
/nix/store/wsv9iz7p87a72blxkj24828vm4ibb9in-python3-3.8.6/lib/python3.8/os.py:213: in makedirs
    makedirs(head, exist_ok=exist_ok)
/nix/store/wsv9iz7p87a72blxkj24828vm4ibb9in-python3-3.8.6/lib/python3.8/os.py:213: in makedirs
    makedirs(head, exist_ok=exist_ok)
/nix/store/wsv9iz7p87a72blxkj24828vm4ibb9in-python3-3.8.6/lib/python3.8/os.py:213: in makedirs
    makedirs(head, exist_ok=exist_ok)
/nix/store/wsv9iz7p87a72blxkj24828vm4ibb9in-python3-3.8.6/lib/python3.8/os.py:223: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 1] Operation not permitted: '/homeless-shelter'
=============================== warnings summary ===============================
tests/bindings/test_processors.py:150
  /private/tmp/nix-build-tokenizers-0.10.0.drv-0/source/bindings/python/tests/bindings/test_processors.py:150: DeprecationWarning: invalid escape sequence \(
    with pytest.raises(Exception, match="Missing SpecialToken\(s\) with id\(s\)"):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
ERROR tests/documentation/test_tutorial_train_from_iterators.py - PermissionE...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 1 warning, 1 error in 2.78s ==========================
builder for '/nix/store/hkbpqdw9f0p77rwy4g1l2yg8in82jp5v-tokenizers-0.10.0.drv' failed with exit code 2
cannot build derivation '/nix/store/zix4mw1ar77xjrjdr66jsx4d8s37vaw0-python3.8-transformers-4.1.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/zix4mw1ar77xjrjdr66jsx4d8s37vaw0-python3.8-transformers-4.1.1.drv' failed
```

File: pkgs/development/python-modules/tokenizers/default.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/tokenizers/default.nix:114]: Thanks, and sorry for the delay (busy week). Should be fixed now.

**Comment by symphorien** [comment]
this breaks python3Packages.transformers. Presumably because of overly restrictive pinning, but as maintainer you probably know better.

**Comment by danieldk** [comment]
> this breaks python3Packages.transformers. Presumably because of overly restrictive pinning, but as maintainer you probably know better.

Indeed, transformers pins tokenizers to a specific version. We do relax the bounds in transformers, but it was too specific. Now also fixed as part of #109505 . Thanks!

**Pull request merged**